src/components/TimeQuerier/index.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from "react";
+import {
+  Text,
+  VStack,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  NumberIncrementStepper,
+  NumberDecrementStepper,
+  useStyleConfig,
+  Box,
+  Button,
+  Alert,
+  AlertIcon,
+} from "@chakra-ui/react";
+import moment from "moment";
+import Datetime from "react-datetime";
+import constants from "../../utils/constants";
+import { TimeQuerierStore } from "../../store/timeQuerier";
+
+const TimeQuerier: React.FC = () => {
+  const styles = useStyleConfig("DateTime", {});
+  const {
+    timeQuerierState,
+    changeTimeQuerier,
+  } = TimeQuerierStore.useContainer();
+  const { defaultStepValue, minStepValue, dateFormat, timeFormat } = constants;
+  const [startTime, setStartTime] = useState(
+    moment(timeQuerierState.selectedStartTimestamp)
+  );
+  const [endTime, setEndTime] = useState(
+    moment(timeQuerierState.selectedEndTimestamp)
+  );
+  const [stepTime, setStepTime] = useState(defaultStepValue);
+  const [error, setError] = useState<undefined | string>(undefined);
+
+  const handleStartChange = (date: moment.Moment | string) => {
+    if (typeof date !== "string") {
+      setStartTime(date);
+      setError(undefined);
+    }
+  };
+  const handleEndChange = (date: moment.Moment | string) => {
+    if (typeof date !== "string") {
+      setEndTime(date);
+      setError(undefined);
+    }
+  };
+  const handleStepChange = (valueAsString: string, valueAsNumber: number) => {
+    if (valueAsString !== "" && valueAsNumber >= minStepValue)
+      setStepTime(valueAsNumber);
+  };
+  const valid = (current: moment.Moment) => {
+    return current.isBefore(moment());
+  };
+  const handleFetchTimeSeriesData = () => {
+    if (
+      endTime.isBefore(startTime) ||
+      endTime.isAfter(moment()) ||
```

Comment:
can we also add a check where the start time is greater than the end time?

Reply:
@muskankhedia  line 58 checks that

Reply:
Can you share a working screenshot of the edge case in the application?

Reply:
Ohh I got that actually when we change the time at that time I don't do the check as I have given freedom at that time because there might be a case when user change the start time first and then end time actually the start time and end time are used for querying once the user has pressed fetch